src/boss/cloudvm/gcp_worker.go

```diff
@@ -97,25 +97,29 @@ func (pool *GcpWorkerPool) CreateInstance(worker *Worker) {
 		fmt.Printf("instance alreay exists!\n")
 		client.startGcpInstance(worker.workerId)
 	} else if err != nil {
-		panic(err)
+		return err
 	}
 
 	lookup, err := client.GcpInstancetoIP()
 	if err != nil {
-		panic(err)
+		return err
 	}
 
 	worker.host = lookup[worker.workerId]
 
 	worker.runCmd("./ol worker up -d")
+
+	return nil
 }
 
-func (pool *GcpWorkerPool) DeleteInstance(worker *Worker) {
+func (pool *GcpWorkerPool) DeleteInstance(worker *Worker) error {
```

Comment:
It seems this never returns an error.  That's actually pretty worrisome because if we don't delete an instance, but the boss thinks it is deleted, the boss could created too many workers, creating a big bill.  I know it's not a bug in your code, so let's leave a TODO to fix this.